epitator/count_annotator_2.py

```diff
@@ -0,0 +1,209 @@
+    stop looking, because we assume that the noun chunk refers to a
+    resultative of an infection event.
+
+    If the noun chunk contains a lemma indicating a person, we continue
+    looking for words in the subtree and ancestors which would indicate that
+    this person was the victim of an infection event.
+
+    Regardless, at the end of that process, we have a list of spaCy words and
+    a list of attribute dicts. These are combined to generate the text span
+    and a merged metadata object, which are used to create a CountSpan and
+    returned.
+
+    TODO: Have an argument flag for "compatibility mode", which would replace
+    all attributes named "infection" with "case".
+
+    If we continue down this path, I'd want to write a class which could take
+    attribute-associated AnnoSpans -- say AttribSpans -- as arguments to event
+    slots.
+    """
+    if not isinstance(nc, SentSpan):
+        raise TypeError("Please give us a SentSpan.")
+
+    # TODO: This should check that nc is actually a noun chunk. It should also
+    # just expect a spaCy noun chunk. I could also see it being better to
+    # accept stuff liberally here, so that it could take any AnnoSpan, check
+    # if it's a noun chunk, and if so, get to work on it.
+    nc_metadata = generate_metadata_for_words(nc.span)
+
+    disjoint_subtree = [w for w in nc.span.subtree if w.i not in [w.i for w in nc.span]]
+    subtree_metadata = generate_metadata_for_words(disjoint_subtree)
+
+    ancestors = [a for a in nc.span.root.ancestors]
+    ancestor_metadata = generate_metadata_for_words(ancestors)
+
+    # Now we use some simple logic to assemble the final set of spans and
+    # metadata for the span.
+    if any([lemma in nc_metadata["attributes"] for lemma in ["infection", "death", "hospitalization"]]):
+        spacy_words = [word for word in nc.span]
+        metadata_dicts = [nc_metadata]
+        sources = ["noun_chunk"]
+    elif "person" in nc_metadata["attributes"]:
+        spacy_words = [word for word in nc.span]
+        metadata_dicts = [nc_metadata]
+        sources = ["noun_chunk"]
+        if any([lemma in subtree_metadata["attributes"] for lemma in ["infection", "death", "hospitalization"]]):
+            # TODO: Consider iterating through until a triggering word is
+            # found.
+            spacy_words.extend(disjoint_subtree)
+            metadata_dicts.append(subtree_metadata)
+            sources.append("subtree")
+        if any([lemma in ancestor_metadata["attributes"] for lemma in ["infection", "death", "hospitalization"]]):
+            spacy_words.extend(ancestors)
+            metadata_dicts.append(ancestor_metadata)
+            sources.append("ancestors")
+    else:
+        return
+
+    start = min([w.idx for w in spacy_words])
+    end = max([w.idx + len(w) for w in spacy_words])
+    anno_span = AnnoSpan(start, end, nc.span.doc)
+
+    merged_metadata = merge_dicts(metadata_dicts, unique=["attributes"], simplify=["count"])
```

Comment:
It would be a good idea to check that there is one and only one count in the merged dict.

Reply:
I am doing this by checking to see if the `count` value is a list (if there are multiple counts, the merge won't have simplified them to a int or double).